corehq/util/validation.py

```diff
@@ -35,7 +35,7 @@ def is_url_or_host_banned(url_or_host):
 
 
 def has_scheme(url):
-    scheme_regex = r'(?:.+:)?//'  # Should match 'http://', 'file://', '//' etc
+    scheme_regex = r'(?:[^:]+:)?//'  # Should match 'http://', 'file://', '//' etc
```

Comment:
Would it be reasonable to also only match from the beginning of the string?
```suggestion
    scheme_regex = r'^(?:[^:]+:)?//'  # Should match 'http://', 'file://', '//' etc
```